File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        return solve(root);
    }
private:
    int solve(TreeNode* root) {
        queue<TreeNode*> queue;
        queue.push(root);
        
        int sum = INT_MIN;
        int level = 0;
        int maxLevel;
        
        while(!queue.empty()) {
            int levelSize = queue.size();
            int levelSum = 0;
            ++level;
            while(levelSize--) {
                auto currNode = queue.front();
                queue.pop();
                levelSum += currNode -> val;
                
                if(currNode -> left != nullptr)
                    queue.push(currNode -> left);
                
                if(currNode -> right != nullptr)
                    queue.push(currNode -> right);
            }
            if(levelSum > sum) {
                sum = levelSum;
                maxLevel = level;
            }
        }
        return maxLevel;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        queue = deque()
        maxSum = -float('inf')
        maxLevel = 0
        level = 0
        
        queue.append(root)
        
        while queue:
            levelSize = len(queue)
            level += 1
            levelSum = 0
            
            for _ in range(levelSize):
                currNode = queue.popleft()
                
                levelSum += currNode.val
                
                if currNode.left != None:
                    queue.append(currNode.left)
                    
                if currNode.right != None:
                    queue.append(currNode.right)
            if levelSum > maxSum:
                maxSum = levelSum
                maxLevel = level
        return maxLevel